injector_helper.c

```diff
@@ -554,10 +554,14 @@ static bool parse_host_uri(const char *uri, char *host, size_t host_max_len, uin
         return false;
     }
 
-    size_t addr_len = MIN((size_t)(addr_end - uri), host_max_len);
+    size_t s = addr_end - uri;
 
-    memcpy(host, uri, addr_len);
-    host[addr_len] = 0;
+    if (s >= host_max_len) {
+        return false;
+    }
+
+    memcpy(host, uri, s);
+    host[s] = '\0';
```

Comment:
Thanks, it's just that I don't think `strlcpy` is appropriate in this case. If we used it the code it would read

```extract the "host" part of this URL into that buffer, truncate it if it's too long```

that is what the previous code tried to do incorrectly and it's where `strlcpy` would have helped. But I think what we really want is 

```extract the "host" part of this URL into that buffer, return failure if it doesn't fit```

which is what this modified code does.